### PR TITLE
Update net core app to 2.0

### PR DESCRIPTION
**Pull request by ericstj**
Version NETCore.App to 2.0 and make it target netcoreapp2.0.

Fix the tests to build without having packages from other RIDs.

Manually add conflict resolution by using `"exclude": "all"` for all packages absorbed into NETCore.App.  This is just a workaround until we can switch to msbuild-based CLI which has the conflict resolution target.

Make NETCore.App depend on NETStandard.Library 2 so that it brings in the conflict resolution targets.

/cc @weshaggard 

This may still fail CI due to myget delay's indexing the new NETStandard.Library 2.0.0-* package.

### CODE REVIEW

**Comment by ellismg** [comment]
> Fix the tests to build without having packages from other RIDs.

Yay!

**Comment by ellismg** [comment]
@dotnet-bot test this please

(Hopefully this should pick up my changes to do archiving)

**Comment by ellismg** [comment]
Appears to be working, @ericstj.  You can pull the artifacts down from details on each job (there should also be a URL to pull down zip of all artifacts for each leg).

**Comment by ericstj** [comment]
Cool, thanks @ellismg

File: pkg/projects/Microsoft.NETCore.App/Microsoft.NETCore.App.pkgproj [hunk 1]
```diff
@@ -51,9 +51,24 @@
...
+      <TargetPath>.NETCoreApp</TargetPath>
+    </File>
+
+    <!-- Bring in Platforms for RID graph, NETStandard.Library for build-tools -->
     <Dependency Include="Microsoft.NETCore.Platforms">
       <Version>1.2.0-beta-24821-0</Version>
-      <TargetFramework>.NETCoreApp1.1</TargetFramework>
+      <TargetFramework>.NETCoreApp2.0</TargetFramework>
+    </Dependency>
+    <Dependency Include="NETStandard.Library">
```
Comment (weshaggard) [inline pkg/projects/Microsoft.NETCore.App/Microsoft.NETCore.App.pkgproj]: Did you publish a newer version of this package with all your changes?

File: pkg/projects/Microsoft.NETCore.App/netcoreapp1.1.props [hunk 2]
```diff
@@ -0,0 +1,131 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <ItemGroup>
```
Comment (weshaggard) [inline pkg/projects/Microsoft.NETCore.App/netcoreapp1.1.props]: Don't we need Microsoft.NETCore.Platforms in this list?

**Comment by weshaggard** [comment]
>Fix the tests to build without having packages from other RIDs.

Is the one template project.json file the only change needed? We should probably also clean up the dummy package generation now, @AlfredoMS.

**Comment by gkhanna79** [comment]
@ericstj As part of bringing up the flat package, can you also look into fixing https://github.com/dotnet/core-setup/issues/736?

CC @weshaggard

File: pkg/projects/Microsoft.NETCore.App/Microsoft.NETCore.App.pkgproj (hunk 1 above)
Comment (ericstj) [inline pkg/projects/Microsoft.NETCore.App/Microsoft.NETCore.App.pkgproj]: Yeah, currently only on dotnet-core-dev-eng.

File: pkg/projects/Microsoft.NETCore.App/netcoreapp1.1.props (hunk 2 above)
Comment (ericstj) [inline pkg/projects/Microsoft.NETCore.App/netcoreapp1.1.props]: It comes in transitively.  Any package that uses RIDs references it.

**Comment by ericstj** [comment]
> Is the one template project.json file the only change needed? 

These were the changes needed: https://github.com/dotnet/core-setup/pull/1088/commits/aba3ad6e41ea8f26b915be0baf6aebdb3a9ac6a9

I suspect this may break maestro, so we may need to fix that to understand project.json.template.  I suspect corefx has the same issue.

**Comment by weshaggard** [comment]
> I suspect corefx has the same issue.

Yes I think it will be broken there as well.

**Comment by dagood** [comment]
@ericstj I don't think anything will break, since those test asset files aren't considered for auto-upgrade. I'm also pretty sure auto-upgrade doesn't care what the runtime keys are, so if/when runtime tokens are added to upgrade-worthy project.jsons I don't know if any changes are needed anyway. If something comes up, I'll definitely help fix it.

(FYI, Maestro is just the service that triggers "things" when a repo changes, so it has no upgrade domain knowledge--that's in buildtools, or core-setup and CLI's custom implementations.)

**Comment by weshaggard** [comment]
@dagood I think the change we will need is for the change commands to also update project.json.template files like they do project.json files. We will need that for our corefx dev/eng work as well.

**Comment by dagood** [comment]
@weshaggard I see, currently buildtools auto-upgrades everything in `ProjectJsonFiles` so it might be doable just to include the `.json.template` files too. The item name gets confusing then, though. Let me know if you'd like me to help test out auto-upgrade on a branch to see if they happen right.

**Comment by weshaggard** [comment]
If it is that simple that will be great and I'm not worried about the property name.

**Pull request merged**